api/src/main/java/org/apache/iceberg/PartitionKey.java

```diff
@@ -18,72 +18,32 @@
  */
 package org.apache.iceberg;
 
-import java.io.Serializable;
-import java.lang.reflect.Array;
-import java.util.Arrays;
 import java.util.List;
-import java.util.function.Function;
-import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
-import org.apache.iceberg.util.SerializableFunction;
+import java.util.stream.Collectors;
 
 /**
  * A struct of partition values.
  *
  * <p>Instances of this class can produce partition values from a data row passed to {@link
  * #partition(StructLike)}.
  */
-public class PartitionKey implements StructLike, Serializable {
+public class PartitionKey extends StructTransform {
```

Comment:
A lot of code has been moved to the non-public `StructTransform` class so that it can be reused for the new `SortKey` class